Xamarin.Forms.Platform.iOS/Platform.cs

```diff
@@ -368,33 +383,16 @@ internal void SetPage(Page newRoot)
 				return;
 			if (Page != null)
 				throw new NotImplementedException();
-			Page = newRoot;
-
-			if (_appeared == false)
-				return;
 
+			Page = newRoot;
 			Page.Platform = this;
 			AddChild(Page);
 
 			Page.DescendantRemoved += HandleChildRemoved;
 
+			_animateModals = false;
 			Application.Current.NavigationProxy.Inner = this;
-		}
-
-		internal void WillAppear()
-		{
-			if (_appeared)
-				return;
-
-			_renderer.View.BackgroundColor = UIColor.White;
-			_renderer.View.ContentMode = UIViewContentMode.Redraw;
-
-			Page.Platform = this;
-			AddChild(Page);
-
-			Page.DescendantRemoved += HandleChildRemoved;
-
-			_appeared = true;
```

Comment:
All of this code can be moved to `Platform` constructor and `SetPage`. Note that `SetPage` never executed below `_appeared = false` because when it's called (there is only one reference to `SetPage`), the flag is always false.